diabetes_regression/evaluate/evaluate_model.py

```diff
@@ -99,7 +99,7 @@
     build_uri = builduri_base + build_id
     run.tag("BuildUri", value=build_uri)
 
-# Paramaterize the matrices on which the models should be compared
+# Parameterize the metrices on which the models should be compared
 # Add golden data set on which all the model performance can be evaluated
 try:
     firstRegistration = False
```

Comment:
With this implementation of binary skip eval or not, we would need to add another switch of something like `FORCE_REGISTRATION`.  There is a case where we want the log output of evaluation, but we want to register the model even if it doesn't pass evaluation criteria.

Reply:
Yeah, I don't like the switch idea as well. It's faster to customize eval.py for my needs rather than play with all of these switches.

Reply:
Understood on the ease of customizing eval.py versus playing with switches.  My thinking is for the use case where you don't want to have to keep modifying the eval.py back and forth for these scenarios:

1. Training on a new dataset with no change in the training code.  The metric will likely be close to the same/equal and in its current form cancel the pipeline.
1. Want to do other tests against inference (i.e. run scoring perf tests, etc)
1. MLOpsPython CI build system itself (i.e. we always want to run the deployment of scoring and not have to clean up the deployments & models every time CI hits to have a green build)

Reply:
It's a shame you can't set YAML-defined variables at queue time. An alternative could be to have these as documented "magic switches", and let users define the variables in the UI tool if desired.

---
In reply to: [371530403](https://github.com/microsoft/MLOpsPython/pull/152#discussion_r371530403) [](ancestors = 371530403)

Reply:
@dtzar I see eval step just cancels run based on metrics from train.py. Skipping the evaluation step would still register a model. I believe too many flags to evaluation steps will just make it more ugly :)

Reply:
There are only two places in the single evaluation file where it calls `run.parent.cancel` and yes the desire is for the model to get registered after eval with this new switch.  This doesn't seem messy to me - am I missing something?

Reply:
Too many checks within the eval steps :) Anyways I have added one per flag as requested.